### PR TITLE
[Libffi] Temporary revert to 3.2.2 to build for aarch64-freebsd

### PR DESCRIPTION
**Pull request by JamesWrigley**
As discussed here: https://github.com/JuliaPackaging/Yggdrasil/pull/10006#discussion_r1885031567

This reverts the following commits:
- "[Libffi] Rebuild for freebsd-aarch64 (#9909)":           670483cc247dba8d14365ff8c001253ca550988d
- "Libffi: Build for aarch64-freebsd (#9782)":              f66e34e1a1469b79237ede7ec283b4460b9e08b3
- "Libffi: New version 3.4.6 (#8275)":                      ff6c5543a8362e0f64c1e215df971d068ca56512
- "[Libffi] Upgrade to v3.4.4 (#5834)":                     cca8463eb26fd4acc357c6cd9f06a7cb1105224d
- ~~"libffi: Stop messing with the install diretory (#3906)": 7629d04b17d885309dd15bcb96e7bc990695a7dd~~

Libffi has a more interesting commit history than Expat so I opted to revert all the commits newer than the change to 3.2.2 to be on the safe side. Interestingly, turns out that the JLL version 3.2.2 is actually building the library version 3.2.1.

EDIT: oops, https://github.com/JuliaPackaging/Yggdrasil/pull/3906 wasn't a version change so that shouldn't have been reverted.

### CODE REVIEW

**Comment by JamesWrigley** [comment]
I'm guessing the compilers have been updated since last time this was built, on aarch64-apple-darwin the build was hitting this error: https://github.com/llvm/llvm-project/issues/72802
I added a patch to fix it.

TL;DR: newer LLVM versions are pickier about the operations allowed in `.cfi` regions and it's not legal to declare labels within them anymore. Same thing happened to Julia: https://github.com/JuliaLang/julia/pull/54634
